### PR TITLE
fix: many reference typegraphql

### PR DESCRIPTION
**Pull request by knapeto**
I found the problem with typegraphql query dialect. GET MANY REFERENCE has different query params

### CODE REVIEW

**Comment by knapeto** [comment]
ping @macrozone

File: packages/dataprovider/README.md [hunk 1]
```diff
@@ -546,3 +546,26 @@ const dataProvider = useDataProvider({
...
 })
 ```
+
+### override mutation operation names due to prisma versions breaking changes
+
+Now you can override mutation calls with your prefixes
+
+```ts
+import { Options } from "@ra-data-prisma/dataprovider";
+import { makePrefixedFullName } from "@ra-data-prisma/dataprovider/lib/utils/makePrefixedFullName";
```
Comment (macrozone) [inline packages/dataprovider/README.md:556]: its probably better to export it from top-level or some barrel-file, othrwise we expose internal folder structure that might change in the future

File: packages/dataprovider/src/types.ts [hunk 2]
```diff
@@ -70,6 +70,8 @@ export type CustomizeInputData = {
   };
 };
 
+type PartialRecord<K extends string | number | symbol, T> = { [P in K]?: T };
```
Comment (macrozone) [inline packages/dataprovider/src/types.ts:73]: seems unused

File: packages/dataprovider/src/utils/makeIntrospectionOptions.ts [hunk 3]
```diff
@@ -9,35 +7,56 @@ import {
...
+        makePrefixedFullName(
+          `${pluralize(camelCase(resource.name))}`,
+          options?.aliasPrefix,
+        ),
       [GET_MANY_REFERENCE]: (resource: Resource) =>
-        prefix(`${pluralize(camelCase(resource.name))}`),
+        makePrefixedFullName(
+          `${pluralize(camelCase(resource.name))}`,
+          options?.aliasPrefix,
+        ),
```
Comment (macrozone) [inline packages/dataprovider/src/utils/makeIntrospectionOptions.ts:59]: unrelated refactorings. its a bit hard to spot the actual change ;-)

File: packages/dataprovider/src/buildWhere.ts [hunk 4]
```diff
@@ -413,6 +421,22 @@ const getFilters = (
...
+    ) {
+      return {
+        [originalKey]: {
+          is: {
+            id: {
+              equals: value,
+            },
+          },
+        },
+      };
```
Comment (macrozone) [inline packages/dataprovider/src/buildWhere.ts:437]: shouln't GET_MANY_REFERENCE get many? because this only works if value is not an array

File: packages/dataprovider/README.md (hunk 1 above)
Comment (knapeto) [reply packages/dataprovider/README.md:556]: Ahh yea you are right!
